tooling/nargo_fmt/src/formatter/statement.rs

```diff
@@ -758,6 +791,42 @@ mod tests {
         2
     }
 }
+";
+        assert_format(src, expected);
+    }
+
+    #[test]
+    fn format_empty_while() {
+        let src = " fn foo() {  while  condition  {   }  } ";
+        let expected = "fn foo() {
+    while condition {}
+}
+";
+        assert_format(src, expected);
+    }
+
+    #[test]
+    fn format_non_empty_while() {
+        let src = " fn foo() {  while  condition  {  1 ; 2  }  } ";
+        let expected = "fn foo() {
+    while condition {
+        1;
+        2
+    }
+}
+";
+        assert_format(src, expected);
+    }
+
+    #[test]
+    fn format_while_with_semicolon() {
+        let src = " fn foo() {  while  condition  {  1 ; 2  };  } ";
+        let expected = "fn foo() {
+    while condition {
+        1;
+        2
+    }
```

Comment:
This looks like it should be a syntax error. Missing the semicolon on `2` implies that we're returning this value whereas a while loop doesn't have a return value.

Reply:
Good point! Well, not a syntax error but maybe a semantic error (Rust says that it expected type `()` but got `i32`).

I just checked and we don't error in `for` nor in `loop`. If it's fine with you I'll capture an issue to fix all of those in one go, separately from this PR.

Reply:
https://github.com/noir-lang/noir/issues/7408